app/src/main/java/de/westnordost/streetcomplete/quests/crossing_type/AddCrossingTypeForm.java

```diff
@@ -0,0 +1,43 @@
+package de.westnordost.streetcomplete.quests.crossing_type;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
+
+public class AddCrossingTypeForm extends ImageListQuestAnswerFragment
+{
+    private final OsmItem[] TYPES = new OsmItem[] {
+            new OsmItem("traffic_signals", R.drawable.crossing_type_signals, R.string.quest_crossing_type_signals),
+            new OsmItem("uncontrolled", R.drawable.crossing_type_zebra, R.string.quest_crossing_type_uncontrolled),
+            new OsmItem("island", R.drawable.crossing_type_island, R.string.quest_crossing_type_island),
+            new OsmItem("unmarked", R.drawable.crossing_type_unmarked, R.string.quest_crossing_type_unmarked)
+    };
+
+    @Override protected int getMaxSelectableItems()
+    {
+        return -1;
+    }
+
+    @Override protected OsmItem[] getItems()
+    {
+        return TYPES;
+    }
+
+    @Override protected int getItemsPerRow()
+    {
+        return 4;
```

Comment:
I am pretty sure that 4 items in a row but texts like "Crossing with traffic-island" will not work, this is too crammed. As (IMO) the picture is of secondary importance, my suggestion is to use a different layout for the image selector here:

`imageSelector.setCellLayout(R.layout.small_image_with_label_at_end);`

... which is like this:
```
[ image ] Crossing with traffic lights
[ image ] Crossing with traffic-island
```
(You need to create this layout, it does not exist yet. The image would be smaller then, so that it doesn't look odd with the one line of text.)

What do you think?